apps/pokedex/pokedex.star

```diff
@@ -52,3 +55,34 @@ def main():
 def random():
     """Return a pseudo-random number in [0, 1)"""
     return time.now().nanosecond / (1000 * 1000 * 1000)
+
+def get_pokemon(id):
+    key = "pokemon_{}".format(id)
+    url = POKEAPI_URL.format(id)
+
+    data = cache.get(key)
```

Comment:
For my own understanding: why not just use the URL as the cache key?

Reply:
We totally could! I started with the URL as the cache key and thought the URL might need to be encoded to avoid any weird character issues so I used the ID. I was going to do the same below in the `get_sprites` method but then realized if we use the ID as the key, we could have an issue where the sprite changed in the API for a given pokemon. So I encoded the URL as a base64 encoded string.